pkg/apis/operator/v1alpha1/types.go

```diff
@@ -63,6 +64,9 @@ type GardenSpec struct {
 
 // RuntimeCluster contains configuration for the runtime cluster.
 type RuntimeCluster struct {
+	// Ingress configures Ingress specific settings for the Garden cluster. This field is immutable.
+	// +optional
+	Ingress *Ingress `json:"ingress,omitempty"`
```

Comment:
This should not be optional, right?

Reply:
Can we use `gardencorev1beta1.Ingress` here?

Reply:
> Can we use `gardencorev1beta1.Ingress` here?

What about this comment?

Reply:
Thought the discussion is between you. What about the already-added CRD [validations](https://github.com/gardener/gardener/blob/351d72b4bb9ab4e9fc8e0907dfa824c8cec52011/pkg/apis/operator/v1alpha1/types.go#L79-L98) via Kubebuilder? 
```
// +kubebuilder:validation:MaxLength=253
// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Value is immutable"
// +kubebuilder:validation:Pattern="^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$"
```
Do I need to move them to `gardencorev1beta1.Ingress`?

`IngressController` supported kind is set to be nginx again by Kubebuilder.
```
// +kubebuilder:validation:Enum="nginx"
```

Reply:
Yes, I think this makes sense.

Reply:
You can extract the validation here https://github.com/gardener/gardener/blob/c5e46d5402ba71f7c95fd04576bbd3f0e0258c5f/pkg/apis/core/validation/seed.go#L203 to a separate function and reuse it for garden.

Reply:
Well, the last comments are contradicting themselves.

If we have CRD validation we may remove the Golang validation. From what I've checked in the code in the garden we don't have `dns.provider`.

Reply:
If we are using `gardencorev1beta1.Ingress`, how do we add CRD validation? Oh, you mean to add it to the `gardencorev1beta1` type. If that is supported, then okay.